### ISSUE TITLE
Unhandled rejection VError: a handler errored, process exiting: ... lib/resultsCapturer.js:22:

### ISSUE THREAD

**Issue by pantherqin** [protractor-cucumber-framework/protractor-cucumber-framework#86]
Hi,

I recently updated protractor-cucumber-framework to …^3.0.0, using protractor 5.1.2, cucumberjs 2.2.0. However I got error "Unhandled rejection VError" as listed in the title. 

What I found out is that if I do the following changes, the above error goes away.

resultsCapturer.js: 
```
this.registerHandler('AfterScenario', function (scenario, callback) {
    ...
    ...
    scenarioFinished.then(function () {
        callback();
    });
    // scenarioFinished.then(callback);
}
```

Please advise if there is anything I can do to eliminate this error without changing anything from protractor-cucumber-framework source code.

Thank you very much.

Chuan

**Comment by darrinholst** [comment]
what's your node version?

**Comment by pantherqin** [comment]
@darrinholst 
It seems all the other "registerHandler" are invoking callback directly as "callback()".

**Comment by darrinholst** [comment]
the difference is that we were preparing for [protractor 6 requirements](https://github.com/angular/protractor/blob/5.1.2/lib/frameworks/README.md#future-requirements), so we're waiting for the `afterEach` promise to resolve if present.

As far as I know `scenarioFinished.then(callback)` is the same as `scenarioFinished.then(function() {callback();})`, but something's different there if that's the only thing you changed.

**Comment by darrinholst** [comment]
what version of q is being used?

**Comment by pantherqin** [comment]
Yes, I used to think .then(foo) and .then(function () { foo() }) behave the same as well..

**Comment by darrinholst** [comment]
are you using `restartBrowserBetweenTests`?

**Comment by pantherqin** [comment]
Yes. Seems that is the reason, or part of the reason. 

> Failing to call runner.afterEach will cause features like restartBrowserBetweenTests to fail. Protractor may also log a warning to the console.

However in the code, we didn't "fail to call" runner.afterEach isn't it? 

scenarioFinished.then === state.runner.afterEach().then in this case.

**Comment by darrinholst** [comment]
`afterEach` is being called [here](https://github.com/protractor-cucumber-framework/protractor-cucumber-framework/blob/master/lib/resultsCapturer.js#L29). If you're restarting your browser then the restart promise is returned [here](https://github.com/angular/protractor/blob/master/lib/runner.ts#L115). It seems that's the promise that's rejecting. I guess we need to handle `afterEach` rejections from protractor?

**Comment by pantherqin** [comment]
afterEach() in protractor is executed before we invoke scenarioFinished.then. In fact scenarioFinished is exactly the promised returned by this.restartPromise, which is assigned [here](https://github.com/angular/protractor/blob/master/lib/runner.ts#L440). 

Do you mean browser_.restart() gives rejection/failure, so we need to handle it?

**Comment by darrinholst** [comment]
> Do you mean browser_.restart() gives rejection/failure, so we need to handle it?

yes

**Comment by darrinholst** [comment]
let me know if 3.1.1 resolves this for you

**Comment by pantherqin** [comment]
PERFECT! Tested and the error is gone. I guess you need to upload it to npm?(I installed through "npm install git+https://github.com/protractor-cucumber-framework/protractor-cucumber-framework.git")

Thank you.

**Comment by darrinholst** [comment]
It was published as version 3.1.1
